Add E2E tests for denying via WP Consent API

tomalec opened the pull request:
### Changes proposed in this Pull Request:

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

This is a small add-on to https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/429.
So, we test that a visitor can also deny their consent using the WP Consent API.

### Checks:
<!-- Mark completed items with an [x] -->
* [x] Does your code follow the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Screenshots:

<!--- Optional --->
![image](https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/17435/ddf4e286-3368-4a39-899b-24fe9fc6e507)


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Checkout `add/wp-consent-api-integration-tests-2` and build extension
2. Setup [E2E environment and run tests](https://github.com/woocommerce/woocommerce-google-analytics-integration?tab=readme-ov-file#e2e-testing)
3. Confirm all tests pass

tomalec committed:
Add E2E tests for denying via WP Consent API

eason9487 commented on tests/e2e/specs/js-scripts/wp-consent-api.test.js:
```diff
@@ -84,6 +84,66 @@ test.describe( 'WP Consent API Integration', () => {
+	test( 'Consent update denying `analytics_storage` is sent when WP Consent API `statistics` category is `denied`', async ( {
+		page,
+	} ) => {
+		await page.goto( 'shop?consent_default=granted' );
```
Might the `consent_default` parameter be omitted? This would be a bit more consistent with this test case:

https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/88f9ea30ac687eccc84248fbeb557081a61b0280/tests/e2e/specs/js-scripts/wp-consent-api.test.js#L184-L191

But if the purpose of specifying this parameter is to explicitly express its initial value, then I think that's also good, and perhaps the parameter could be added to the last test case as well.

tomalec replied:
Thanks, I removed it for consistency :)

eason9487 reviewed (approved):
Thanks for adding more test cases. I can run all E2E tests successfully. LGTM.

The only comment is not a blocker

tomalec committed:
Rely on the default granted test setup.

Address https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/430/files/88f9ea30ac687eccc84248fbeb557081a61b0280#r1625282895
